secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  RecoveryPhraseValidation.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import Foundation
+import ComposableArchitecture
+import SwiftUI
+
+typealias RecoveryPhraseValidationStore = Store<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+typealias RecoveryPhraseValidationViewStore = ViewStore<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+
+/// Represents the data of a word that has been placed into an empty position, that will be used
+/// to validate the completed phrase when all ValidationWords have been placed.
+struct ValidationWord: Equatable {
+    var groupIndex: Int
+    var word: String
+}
+
+struct RecoveryPhraseValidationState: Equatable {
+    enum Step: Equatable {
+        case initial
```

Comment:
Because this can traverse from one state to another and they don't happen sequentially, I would use Status here for the name of your enum. Step implies that you will sequentially move through all of these, which could happen, but isn't required. In Onboarding, you sequentially move through steps either forward or backward, so this name applies better there, where here you can move to any state at any time, so this represents more of the current status of validating the phrase then it does an individual step.

Reply:
Strictly on the semantic meaning of the word is ok. But status is often use to denote states of a thing which is not directly actionable by the user. (like the connection status) You inquiry the status of something that's handled by a third party which owns the state of the process and informs the status. in this case the process is quite self actionable so the correct word for it as it is often used in software engineering would be State, but it's already in use by the encapsulating Struct :S

Reply:
I don't think the word status has any implications on whom is responsible for editing it.

Reply:
That being said, we do only have 3 cases in this enum now.  We had more before, that allowed for more non-sequential flow here.  With these 3 now, I think you can only move through them in order, forward or back, so over time I think this enum has turned into something that better suits the current name than what we originally had.  👍 We can leave it! 🙌